src/app/utils/string-validators.ts

```diff
@@ -0,0 +1,63 @@
+export function containNameCharactersOnly(str: string): boolean {
+	const testRegExp: RegExp = /^([a-zA-Z0-9_]+)$/;
+	const match: RegExpExecArray = testRegExp.exec(str);
+	return (match) ? true : false;
+}
+
+export function containAlphabetsOnly(str: string): boolean {
+	const testRegExp: RegExp = /^([a-zA-Z]+)$/;
+	const match: RegExpExecArray = testRegExp.exec(str);
+	return (match) ? true : false;
+}
+
+export function isValidApiAcceptedDateString(str: string): boolean {
+	if (str === '') {
+		return false;
+	}
+
+	const testRegExp = /^(\d{4})-(\d{2})-(\d{2})_(\d{2}):(\d{2})$/;
+	const match: RegExpExecArray = testRegExp.exec(str);
+
+	if (!match) {
+		return false;
+	}
+
+	const parsedYear = parseInt(match[1], 10);
+	const parsedMonth = parseInt(match[2], 10);
+	const parsedDate = parseInt(match[3], 10);
+	const parsedHour = parseInt(match[4], 10);
+	const parsedMinute = parseInt(match[5], 10);
+
+	if (parsedHour < 0 || parsedHour >= 24) {
+		return false;
+	}
+	if (parsedMinute < 0 || parsedMinute >= 60) {
+		return false;
+	}
+
+	if (parsedYear < 0) {
```

Comment:
I am not sure if this can be done. Since all calculations are based on 1 january 1970, lower limit can be 1970.